core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -206,9 +236,15 @@ private[coordinator] class GroupMetadata(val groupId: String, val protocolType:
 
   def initNextGeneration() = {
     assert(notYetRejoinedMembers == List.empty[MemberMetadata])
-    generationId += 1
-    protocol = selectProtocol
-    transitionTo(AwaitingSync)
+    if (!isEmpty) {
+      generationId += 1
+      protocol = selectProtocol
+      transitionTo(AwaitingSync)
+    } else {
+      generationId += 1
+      protocol = null
+      transitionTo(Empty)
```

Comment:
so now there's an Empty state, a GroupMetadata.isEmpty, and a GroupMetadata.is(Empty). Does it make sense to get rid of isEmpty?